config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml

```diff
@@ -97,3 +97,17 @@ presubmits:
         - --warnings=duplicate-job-refs
     annotations:
       testgrid-dashboards: presubmits-test-infra
+
+  - name: pull-test-infra-verify-golangci-lint
+    decorate: true
+    run_if_changed: '\.go'
+    spec:
+      containers:
+        - image: golangci/golangci-lint:v1.31.0
```

Comment:
Is this in sync with the version used via bazel and the repo-infra route? ref: https://github.com/kubernetes/test-infra/issues/19588#issuecomment-709593545

I'm guessing @fejta would rather see us upgrade it in repo-infra

Reply:
What I don't understand is the value of pulling this in via repo infra. We want to specify a version of this we use, not get a specific version pulled in transiently by a dependency. 

Additionally, linting needs a lot of ram, downstream we currently request 10 gig. Coupling this with the generic bazel job means that we need to request as much for the bazel job, even though its not needed for most of its runtime.

Reply:
@spiffxp I'm more than happy to have a go at upgrading golang-ci in repo-infra, but not sure on the knock effect this will have...

Reply:
Upgrading golangci-lint in repo-infra is not a problem

Ideally everything needed to verify the repo happens inside `bazel test //...`

Ideally we have a single version of golangci-lint to worry about across the org.

Reply:
@fejta can you either provide guidance on how to continue here with bazel or let us add the job outside of it?

FWIW, I don't see any value in having the same version of a linter across the org. Everyone should be able to update at their own pace.